sklearn/utils/stats.py

```diff
@@ -1,18 +1,61 @@
 import numpy as np
 
 from .extmath import stable_cumsum
+from .fixes import _take_along_axis
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
+    """Compute weighted percentile
+
+    Computes lower weighted percentile. If `array` is a 2D array, the
+    `percentile` is computed along the axis 0.
+
+        .. versionchanged:: 0.24
+            Accepts 2D `array`.
+
+    Parameters
+    ----------
+    array : 1D or 2D array
+        Values to take the weighted percentile of.
+
+    sample_weight: 1D or 2D array
+        Weights for each value in `array`. Must be same shape as `array` or
+        of shape `(array.shape[0],)`.
+
+    percentile: int, default=50
+        Percentile to compute. Must be value between 0 and 100.
+
+    Returns
+    -------
+    percentile : int if `array` 1D, ndarray if `array` 2D
+        Weighted percentile.
     """
-    Compute the weighted ``percentile`` of ``array`` with ``sample_weight``.
-    """
-    sorted_idx = np.argsort(array)
+    n_dim = array.ndim
+    if n_dim == 0:
+        return array[()]
+    if array.ndim == 1:
+        array = array.reshape((-1, 1))
+    # When sample_weight 1D, repeat for each array.shape[1]
+    if (array.shape != sample_weight.shape and
+            array.shape[0] == sample_weight.shape[0]):
```

Comment:
Not sure of formatting of this (@glemaitre )

Reply:
That's good